tests/test_hls_report.py

```diff
@@ -108,102 +97,155 @@ def refine(res_tbl):
     res[0] = res[0].replace(', ', '', 1)
     return res
 
-def get_expected(wd):
+def get_expected(ver, wd):
     path = pathlib.Path(__file__).parent.absolute()
     path = str(path) + '/test_report_data/expected.json'
     with open(path) as f:
         data = json.loads(f.read())
-    return data[wd]
+    return data[ver][wd]
 
-def test_col(vhls):
-    if vhls:
-        rpt = sobel()
-    else:
-        rpt = parse_rpt()
-    res = rpt.display()
-    lst = refine(res)
-    assert lst[0] == get_expected("Category")
-
-def test_info(vhls):
-    if vhls:
-        rpt = sobel()
-    else:
-        rpt = parse_rpt()
-    res = rpt.display()
-    lst = refine(res)
-    assert lst == get_expected("NoQuery")
-
-def test_loop_query(vhls):
-    if vhls:
-        rpt = sobel()
-    else:
-        rpt = parse_rpt()
-    row_query = ['P', 'A']
-    lq = rpt.display(loops=row_query)
-    lq_lst = refine(lq)
-    assert lq_lst == get_expected("LoopQuery")
-
-def test_column_query(vhls):
-    if vhls:
-        rpt = sobel()
-    else:
-        rpt = parse_rpt()
-    col_query = ['Trip Count', 'Latency', 'Iteration Latency', 
-                 'Pipeline II', 'Pipeline Depth']
-    cq = rpt.display(cols=col_query)
-    cq_lst = refine(cq)
-    assert cq_lst == get_expected("ColumnQuery")
-
-def test_level_query(vhls):
-    if vhls:  
-        rpt = sobel()
-    else:
-        rpt = parse_rpt()
-    lev_query = 1
-    vq = rpt.display(level=lev_query)
-    vq_lst = refine(vq)
-    assert vq_lst == get_expected("LevelQuery")
-
-def test_level_oob_query(vhls):
-    if vhls:
-        rpt = sobel()
-    else:
-        rpt = parse_rpt()
-    lev_query = 5
-    vq = rpt.display(level=lev_query)
-    vq_lst = refine(vq)
-    assert vq_lst == get_expected("LevelQueryOOB")
-    lev_query = -2
-    try:
-        vq = rpt.display(level=lev_query)
-    except IndexError:
-        assert True
-    return
-
-def test_multi_query(vhls):
-    if vhls:
-        rpt = sobel()
-    else:
-        rpt = parse_rpt()
-    row_query = ['P', 'A']
-    lev_query = 1
-    mq = rpt.display(loops=row_query, level=lev_query)
-    mq_lst = refine(mq)
-    assert mq_lst == get_expected("MultiQuery")
-
-def test_all_query(vhls):
-    if vhls:
-        rpt = sobel()
+def parse_rpt(alg):
+    path = pathlib.Path(__file__).parent.absolute()
+
+    if alg == 'sobel':
+        xml_file = str(path) + '/test_report_data/sobel_report.xml'
     else:
-        rpt = parse_rpt()
-    row_query = ['P', 'A']
-    col_query = ['Latency']
-    lev_query = 1
-    aq = rpt.display(loops=row_query, level=lev_query, cols=col_query)
-    aq_lst = refine(aq)
-    assert aq_lst == get_expected("AllQuery")
+        xml_file = str(path) + '/test_report_data/spam_filter_report.xml'
```

Comment:
Can we just put the path inside the config. I don't like `if` `else` here. If we have more, we will have long if/else cases.

Reply:
True.